test/unit/auth/token-verifier.spec.ts

```diff
@@ -513,4 +526,305 @@ describe('FirebaseTokenVerifier', () => {
+        .rejects(new JwtError(JwtErrorCode.TOKEN_EXPIRED, 'Expired token.'));
+      stubs.push(verifierStub);
+
+      const mockAuthBlockingToken = mocks.generateAuthBlockingToken();
+
+      return authBlockingTokenVerifier._verifyAuthBlockingToken(mockAuthBlockingToken, false, undefined)
+        .should.eventually.be.rejectedWith(
+          'Firebase Auth Blocking token has expired. Get a fresh Auth Blocking token from your client ' +
+          'app and try again (auth/auth-blocking-token-expired)')
+        .and.have.property('code', 'auth/auth-blocking-token-expired');
+    });
+
+    it('should be rejected when the verifier throws invalid signature for a Auth Blocking JWT token.', () => {
+      const verifierStub = sinon.stub(PublicKeySignatureVerifier.prototype, 'verify')
+        .rejects(new JwtError(JwtErrorCode.INVALID_SIGNATURE, 'invalid signature.'));
+      stubs.push(verifierStub);
+
+      const mockAuthBlockingToken = mocks.generateAuthBlockingToken();
+
+      return authBlockingTokenVerifier._verifyAuthBlockingToken(mockAuthBlockingToken, false, undefined)
+        .should.eventually.be.rejectedWith('Firebase Auth Blocking token has invalid signature');
+    });
+
+    it('should be rejected when the verifier throws key fetch error.', () => {
+      const verifierStub = sinon.stub(PublicKeySignatureVerifier.prototype, 'verify')
+        .rejects(new JwtError(JwtErrorCode.KEY_FETCH_ERROR, 'Error fetching public keys.'));
+      stubs.push(verifierStub);
+
+      const mockAuthBlockingToken = mocks.generateAuthBlockingToken();
+
+      return authBlockingTokenVerifier._verifyAuthBlockingToken(mockAuthBlockingToken, false, undefined)
+        .should.eventually.be.rejectedWith('Error fetching public keys.');
+    });
+
+    it('should be rejected given a custom token with error using article "an" before JWT short name', () => {
+      return tokenGenerator.createCustomToken(mocks.uid)
+        .then((customToken) => {
+          return authBlockingTokenVerifier._verifyAuthBlockingToken(customToken, false, undefined)
+            .should.eventually.be.rejectedWith(
+              '_verifyAuthBlockingToken() expects an Auth Blocking token, but was given a custom token');
+        });
+    });
+
+    it('should be rejected given a custom token with error using article "a" before JWT short name', () => {
+      const tokenVerifierSessionCookie = new verifier.FirebaseTokenVerifier(
+        'https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys',
+        'https://session.firebase.google.com/',
+        verifier.SESSION_COOKIE_INFO,
+        app,
+      );
+      return tokenGenerator.createCustomToken(mocks.uid)
+        .then((customToken) => {
+          return tokenVerifierSessionCookie._verifyAuthBlockingToken(customToken, false, undefined)
+            .should.eventually.be.rejectedWith(
+              'verifySessionCookie() expects a session cookie, but was given a custom token');
```

Comment:
Does `_verifyAuthBlockingToken()` expects a session cookie here?

Reply:
Ahh, I copied this test from verifyIdToken, removed it in the latest push